cli/azd/pkg/tools/azcli/spring_app.go

```diff
@@ -0,0 +1,250 @@
+package azcli
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/appplatform/armappplatform"
+	"github.com/Azure/azure-storage-file-go/azfile"
+	azdinternal "github.com/azure/azure-dev/cli/azd/internal"
+	"github.com/azure/azure-dev/cli/azd/pkg/account"
+	"github.com/azure/azure-dev/cli/azd/pkg/httputil"
+	"log"
+	"net/url"
+	"os"
+)
+
+// SpringService provides artifacts upload/deploy and query to Azure Spring Apps (ASA)
+type SpringService interface {
+	// Get Spring app properties
+	GetSpringAppProperties(
+		ctx context.Context,
+		subscriptionId string,
+		resourceGroupName string,
+		instanceName string,
+		appName string,
+	) (*SpringAppProperties, error)
+	// Deploy jar artifact to ASA app deployment
+	DeploySpringAppArtifact(
+		ctx context.Context,
+		subscriptionId string,
+		resourceGroup string,
+		instanceName string,
+		appName string,
+		relativePath string,
+		deploymentName string,
+	) (*string, error)
+	// Upload jar artifact to ASA app Storage File
+	UploadSpringArtifact(
+		ctx context.Context,
+		subscriptionId string,
+		resourceGroup string,
+		instanceName string,
+		appName string,
+		artifactPath string,
+	) (*string, error)
+	// Get Spring app deployment
+	GetSpringAppDeployment(
+		ctx context.Context,
+		subscriptionId string,
+		resourceGroupName string,
+		instanceName string,
+		appName string,
+		deploymentName string,
+	) (*string, error)
+}
+
+type springService struct {
+	credentialProvider account.SubscriptionCredentialProvider
+	httpClient         httputil.HttpClient
+	userAgent          string
+}
+
+// Creates a new instance of the NewSpringService
+func NewSpringService(
+	credentialProvider account.SubscriptionCredentialProvider,
+	httpClient httputil.HttpClient,
+) SpringService {
+	return &springService{
+		credentialProvider: credentialProvider,
+		httpClient:         httpClient,
+		userAgent:          azdinternal.MakeUserAgentString(""),
+	}
+}
+
+type SpringAppProperties struct {
+	Fqdn []string
+}
+
+func (ss *springService) GetSpringAppProperties(
+	ctx context.Context,
+	subscriptionId, resourceGroup, instanceName, appName string,
+) (*SpringAppProperties, error) {
+	client, err := ss.createSpringAppClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	springApp, err := client.Get(ctx, resourceGroup, instanceName, appName, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed retrieving spring app properties: %w", err)
+	}
+
+	return &SpringAppProperties{
+		Fqdn: []string{*springApp.Properties.Fqdn},
```

Comment:
Can `Fqdn` be null/nil? We've ran into this for other hosts. Prefer if we do a nil-check here.